test/e2e/md_remediation_test.go

```diff
@@ -60,6 +60,9 @@ var _ = Describe("When testing MachineDeployment remediation", func() {
 		// Setup a Namespace where to host objects for this spec and create a watcher for the namespace events.
 		namespace, cancelWatches = setupSpecNamespace(ctx, specName, bootstrapClusterProxy, artifactFolder)
 
+		// Create LXC secret for LXD provider if needed
+		createLXCSecret(ctx, bootstrapClusterProxy, e2eConfig, namespace.Name)
```

Comment:
Well ultimately I don't like having this repeat for every single test although I don't see a quicker or easier alternative, maybe something with webhooks can be created to inject this in some other way after namespace creation. No action needed for now.